### DISCUSSION TITLE
Can "Ref" object get bloated/big like Redux Stores?

### DISCUSSION THREAD

**Discussion by salihgueler** [rrousselGit/riverpod#1399] [Q&A] [answered]
Hello Remi and all the maintainers,

I have been a fan of Provider and with my new side project I wanted to give it a go with riverpod. Great documentation and package so far.

While I was reading the documentation, I could not understand one thing. As you know, Redux keeps the state on a Store and as the app grows, I have seen that the object itself (no matter split that you do), getting bloated and consuming a lot of memory to maintain later on. 

I am seeing ref object to be used throughout the application. My question/concern is, Does the state manupilation happen like the Redux Store? If not, how do you prevent Ref object to be bloated while controlling your state?  If my concerns or questions are invalid, can you direct me to a documentation or a code-piece to understand "Ref" better?

**Comment by rrousselGit** [comment, marked as answer]
Ref is unrelated to this discussion I would say. It's only an interface for allowing a provider to interact with other providers.


But to answer your question: In Riverpod we don't have "one big object". We have multiple small ones.
Each provider is a unique state, and when the provider is disposed nothing remains in the RAM.

As long as you use autoDispose, there should be no problen.

**Reply by msarkrish** [reply]
@rrousselGit in one of my application I'm using future provider with List of Map. Some users will have almost 800 maps and I'm not using autodispose, because my requirement is loading state need to be displayed in first time only, If user navigating to other screen and navigate back to same screen then data need to displayed without loading again.
Whether this will create memory issue or any other problem?
If my code will create problem then what is the solution is correct for problem?

**Reply by rrousselGit** [reply]
> Some users will have almost 800 maps and I'm not using autodispose, because my requirement is loading state need to be displayed in first time only,

Try the v2 and still use autoDispose.

You should see loading the first time only, but this will correctly dispose things too

**Reply by msarkrish** [reply]
Thank you Remi, I will try V2
